### PR TITLE
Cp 18

### PR DESCRIPTION
**Pull request by haseeb413**
updateSiteContactInfo API

Address: should be generated from Google

Company Phones:

 Min=10 digits Max= 15 digits.

Fax: Max= 15 digits

Email:

There should be the @ symbol and a dot.

cannot be more than 255 characters in length.

hyphens cannot be the first or last character.

cannot be all Numerics.

should consist of any combination of alphabetic characters, digits and hyphens.

### CODE REVIEW

File: spring-boot-angular-15-crud-example-master/spring-boot-server/src/main/java/com/bezkoder/spring/jpa/h2/controller/ContactUsController.java [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+    @GetMapping
+    public ResponseEntity<List<ContactUsDto>> getAllContactUs() {
+        List<ContactUsDto> contactUsDTOList = contactUsService.getAllContactUs();
+        return ResponseEntity.ok(contactUsDTOList);
+    }
+
+    @DeleteMapping("/{id}")
+    public ResponseEntity<Void> deleteContactUs(@PathVariable Long id) {
+        contactUsService.deleteContactUs(id);
```
Comment (Mushaf-Cyber) [inline spring-boot-angular-15-crud-example-master/spring-boot-server/src/main/java/com/bezkoder/spring/jpa/h2/controller/ContactUsController.java:38]: in delete you can send a message like contact us deleted successfully instead on no content

File: spring-boot-angular-15-crud-example-master/spring-boot-server/src/main/java/com/bezkoder/spring/jpa/h2/controller/SqliteDialect.java
Comment (Mushaf-Cyber) [inline spring-boot-angular-15-crud-example-master/spring-boot-server/src/main/java/com/bezkoder/spring/jpa/h2/controller/SqliteDialect.java:1]: please remove this file

File: spring-boot-angular-15-crud-example-master/spring-boot-server/src/main/java/com/bezkoder/spring/jpa/h2/dto/ContactUsDto.java [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+    public void setEmail(String email) {
+        this.email = email;
+    }
+
+    public void setPhonenumber(String phonenumber) {
+        this.phonenumber = phonenumber;
+    }
+
+    public void setMessage(String message) {
+        this.message = message;
```
Comment (Mushaf-Cyber) [inline spring-boot-angular-15-crud-example-master/spring-boot-server/src/main/java/com/bezkoder/spring/jpa/h2/dto/ContactUsDto.java:93]: use lombok getters and setters to reduce this code in all classes

File: spring-boot-angular-15-crud-example-master/spring-boot-server/src/main/java/com/bezkoder/spring/jpa/h2/dto/SiteContactInfoDto.java [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+    }
+
+    public void setCompany_Phones(String company_Phones) {
+        Company_Phones = company_Phones;
+    }
+
+    public void setFax(String fax) {
+        Fax = fax;
+    }
+
```
Comment (Mushaf-Cyber) [inline spring-boot-angular-15-crud-example-master/spring-boot-server/src/main/java/com/bezkoder/spring/jpa/h2/dto/SiteContactInfoDto.java:71]: here also lombok getters and setters

File: spring-boot-angular-15-crud-example-master/spring-boot-server/src/main/java/com/bezkoder/spring/jpa/h2/service/SiteContactInfoServiceImpl.java [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+    @Override
+    public SiteContactInfoDto getSiteContactInfo() {
+        // Implement custom logic to fetch the SiteContactInfo entity
+        // For example, you can use JpaRepository methods like findById()
+        // or custom JPQL queries to retrieve the entity from the repository
+
+        // Example using JpaRepository method findById()
+        Long contactInfoId = 1L; // Id of the contact info entity to fetch
+        Optional<SiteContactInfo> contactInfoOptional = siteContactInfoRepository.findById(contactInfoId);
+
```
Comment (Mushaf-Cyber) [inline spring-boot-angular-15-crud-example-master/spring-boot-server/src/main/java/com/bezkoder/spring/jpa/h2/service/SiteContactInfoServiceImpl.java:29]: what use of this method you passing id 1L predefined here?

**Review by Mushaf-Cyber — CHANGES_REQUESTED** [review]
Added some callouts

File: spring-boot-angular-15-crud-example-master/spring-boot-server/src/main/java/com/bezkoder/spring/jpa/h2/service/SiteContactInfoServiceImpl.java (hunk 4 above)
Comment (haseeb413) [reply spring-boot-angular-15-crud-example-master/spring-boot-server/src/main/java/com/bezkoder/spring/jpa/h2/service/SiteContactInfoServiceImpl.java:29]: In ContactInfo We have only 2 methods Updating and getting so We have only single Contact Information Here I am getting the  contactInfo By Id which we are calling  in Direct ServiceImpl.

File: spring-boot-angular-15-crud-example-master/spring-boot-server/src/main/java/com/bezkoder/spring/jpa/h2/controller/ContactUsController.java (hunk 1 above)
Comment (haseeb413) [reply spring-boot-angular-15-crud-example-master/spring-boot-server/src/main/java/com/bezkoder/spring/jpa/h2/controller/ContactUsController.java:38]: Done.

File: spring-boot-angular-15-crud-example-master/spring-boot-server/src/main/java/com/bezkoder/spring/jpa/h2/controller/SqliteDialect.java
Comment (haseeb413) [reply spring-boot-angular-15-crud-example-master/spring-boot-server/src/main/java/com/bezkoder/spring/jpa/h2/controller/SqliteDialect.java:1]: Done.

**Pull request merged**